testsuite/capabilities.py

```diff
@@ -25,3 +25,21 @@ def has_kuadrant():
                 return False, f"Cluster {cluster.api_url} does not have Kuadrant resource in project {project}"
 
     return True, None
+
+
+@functools.cache
+def kuadrant_version():
+    """Returns catalog image tag of deployed Kuadrant if possible."""
+    clusters = weakget(settings)["control_plane"]["additional_clusters"] % []
+    clusters.insert(0, settings["cluster"])
+    versions = []
+    for cluster in clusters:
+        project = cluster.change_project("openshift-marketplace")
+        if not project.connected:
+            break
```

Comment:
Why break when a catalog or project is not found? For consistency's sake I would expect that we have relevant information in all reports.

Reply:
When the catalog source under exact name `CatalogSource/kuadrant-upstream` inside `openshift-marketplace` namespace is not found than no version is printed. This is intended. I plan in the future to make the search for version better to work in more circumstances - mainly on Kind. 

This feature should be taken just as convince not as production ready and documented feature of the testsuite.

Reply:
Please create an issue for this. Otherwise we forgot about it